### ISSUE TITLE
Helix Work Item Friendly names from Arcade SDK can include non-URL-friendly file names

### ISSUE THREAD

**Issue by MattGal** [dotnet/arcade#3602]
@dougbu fyi.

I noticed this while looking at logs for this run:
https://github.com/aspnet/AspNetCore-Internal/issues/2936

Specifically the Friendly name of the work item was Sockets.FunctionalTests/netcoreapp3.0 which makes the resultant log URI https://helix.dot.net/api/2018-03-14/jobs/4f248f91-1277-4ba5-bf3b-21f0ec0da2e8/workitems/Sockets.FunctionalTests/netcoreapp3.0/console which doesn't work (because it ends up looking for a route ending with Sockets.FunctionalTests and fails.)

We might be able to fix the console log API for Helix to re-hydrate such paths but previous efforts to do this (as well as to handle 404s properly) have been disastrous.

So to summarize, I'd suggest that we either:
- Cause non-URL-friendly Friendly names to cause build failure (which will encourage folks to fix this up in their builds when Arcade updates 
OR
- Coerce the friendly name of work items at queue time to a friendly name, e.g. 
**Sockets.FunctionalTests/netcoreapp3.0 -> Sockets.FunctionalTests-netcoreapp3.0**

(@dougbu  - if it's easily done it'd be great to just change the / to - or something until this is resolved, for log readability reasons)

**Comment by markwilkie** [comment]
cc/ @Chrisboh for visibility

**Comment by ChadNedzlek** [comment]
I'm pretty sure this just need to be escaped before we make up a URL out of the blue... the "/" should turn into %2F that would, I assume, work. We just need to properly URL escape things.

**Comment by dougbu** [comment]
@ChadNedzlek most HTTP servers treat '%2F' and '/' in the request's absolute path identically

**Comment by ChadNedzlek** [comment]
True... this should be... interesting to fix. :-)

**Comment by MattGal** [comment]
> True... this should be... interesting to fix. :-)

This is why I think we should just forcibly change the friendly names in the arcade sender, and warn when we do.  It gets to print and form its own URIs, and the user isn't surprised by the name tweaks because it's in every build log.

**Comment by ChadNedzlek** [comment]
I'll make the change that swaps out / for something else in a jiffy.  '/' is, to my knowledge, the only character ASP.NET can't handle here, the rest will work fine (as long as they are properly encoded)

**Comment by markwilkie** [comment]
@ChadNedzlek - you mentioned that you'd make a change, but also kicked this out of the epic.  Do you think we should we do this or not?  (if not now, then we should close it)

**Comment by ChadNedzlek** [comment]
If there aren't currently any instances of this, I'd rather just make it an error.  Changing things behind people's backs can confuse some people.  Probably just a small check in the arcade SDK to detect this and make it an error.

We can pop it back into our epic, and we can just consider it a weird little thing we are doing... it doesn't really have anything to do with our epic, but it doesn't really have anything to do with any epic.

**Comment by MattGal** [comment]
> If there aren't currently any instances of this, I'd rather just make it an error. 

Pretty sure everyone who cares has sanitized their names now so they can view their results.  Further, if you make it an error and it does matter, the Maestro PR that brings the change will fail and make them get right.  Therefore I say "let's do that".

**Comment by markwilkie** [comment]
Thanks @ChadNedzlek and @mmitche.  Adding back to the epic.

**Comment by ChadNedzlek** [comment]
This doesn't seem super related to the scope of our current epic, back to triage.10073

**Comment by MattGal** [comment]
my two cents: Just making this an error is very much in the scope of FR and will improve user experience.

**Comment by garath** [comment]
Triage: Have arcade throw if not found

**Comment by MattGal** [comment]
After some reading, we've already done the relevant work here so I'll close this:

- [Turn all slashes to dashes so they don't confuse path construction] (https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Helix/Sdk/Helpers.cs#L16)
- [Escape URIs when using work item names in them](https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Helix/Sdk/GetFailedWorkItems.cs#L55)

As such, anything left after this would not be fixed in arcade, but in the code producing uris.  Closing.
